### ISSUE TITLE
[FR] - Pretty printer for Tx

### ISSUE THREAD

**Issue by locallycompact** [IntersectMBO/cardano-api#707]
We have a simple pretty printer for Tx in hydra.

https://github.com/cardano-scaling/hydra/blob/2b20fb7b53b3df37ae9607b7acbd18627f260d6e/hydra-cardano-api/src/Hydra/Cardano/Api/Pretty.hs

We want to factor this out to its own library, perhaps this could be ported to `cardano-api` under a sublibrary such as `cardano-api:pretty`. Our printer just has a `Tx -> String` interface, but we could also leverage something else like `formatting`.

Would this go in this repository? Or somewhere else.

**Comment by locallycompact** [comment]
@Jimbo4350 Is this something you would have an opinion on?

**Comment by carbolymer** [comment]
Nice! That would be useful to have. Does it have to a sublibrary? This code is pretty self contained so I think it can be simply a module in `cardano-api`.

**Comment by locallycompact** [comment]
Yes, sorry I failed to notice all the already existing render functions throughout this codebase so I can just put this somewhere that feels consistent.

**Comment by carbolymer** [comment]
@locallycompact are you still interested in checking in your prettyprinting code?

**Comment by locallycompact** [comment]
I am still interested in doing this but it's low priority at the moment.
